src/rgen/Microsoft.Macios.Generator/AttributesNames.cs

```diff
@@ -15,4 +20,19 @@ public static class AttributesNames {
 	public const string UnsupportedOSPlatformAttribute = "System.Runtime.Versioning.UnsupportedOSPlatformAttribute";
 	public const string ObsoletedOSPlatformAttribute = "System.Runtime.Versioning.ObsoletedOSPlatformAttribute";
 
+	public static string? GetBindingTypeAttributeName<T> () where T : Enum
+	{
+		var type = typeof (T);
+		if (type == typeof (ObjCBindings.Category)) {
+			return BindingCategoryAttribute;
+		}
+		if (type == typeof (ObjCBindings.Class)) {
+			return BindingClassAttribute;
+		}
+		if (type == typeof (ObjCBindings.Protocol)) {
+			return BindingProtocolAttribute;
+		}
+
+		return null;
```

Comment:
Not entirely sure which one I prefer tbh (I lean just a tiny bit on to the switch but not for much), you make the call, I am fine either way.

```suggestion
		var type = typeof (T);
		switch (type) {
			case Type _ when type == typeof (ObjCBindings.Field):
				return ExportFieldAttribute;
			case Type _ when type == typeof (ObjCBindings.Property):
				return ExportPropertyAttribute;
			case Type _ when type == typeof (ObjCBindings.Method):
				return ExportMethodAttribute;
			default:
				return null;
		}
```

Reply:
the one with the switch to be looks very verbose and a switch become an if at the end of the day with a little better branching: https://sharplab.io/#v2:CYLg1APgAgTAjAWAFBQMwAJboMLoN7LpHqHFqZwAMA/OgCoCmAzgC7oAUAlgHZvcCuAWwBGDAE4BKfKWLFOAMw4CR49AF416SlIJJZ+sgHZ0AIgBe4gPYmA3DP0Bfe7OdzF7ZaLHrNcHa4MoYxNLbgZbAPQnPQMSGMDjAQAbJLt4qPtncigqWkZWOA4ePiEvf3TZJgB3ThYAYwALJVLxctj9OoBDJgYtEEj2oNMLMWs09tkunvQ4foqJoiGQsIj5g2AGeU7+JJY5hYT0ZNTI6PtohyA=

I don't know if it is worth the change.

Reply:
I am fine either way so not really pushing for a change 👍